openair2/RIC_AGENT/E2AP/e2ap_generate_messages.c

```diff
@@ -112,55 +112,58 @@ int e2ap_generate_e2_setup_request(ranid_t  ranid,
     oid->buf = (uint8_t *)calloc(oid_len, sizeof(uint8_t));
     memcpy(oid->buf, func->model->oid, oid_len);
     oid->size = oid_len;
-    ran_function_item_ie->value.choice.RANfunction_Item.ranFunctionOID = oid;
+    ran_function_item_ie->value.choice.RANfunction_Item.ranFunctionOID = *oid;
 
     ASN_SEQUENCE_ADD(&ie->value.choice.RANfunctions_List.list,
-		     ran_function_item_ie);
+             ran_function_item_ie);
   }
   ASN_SEQUENCE_ADD(&req->protocolIEs.list,ie);
 
-#if 1
   /*to be re-done during E2AP 2.0.1 integration */
   /* E2nodeComponentConfigUpdate_List */
   E2AP_E2nodeComponentID_t *e2NodeCompId;
   ie = (E2AP_E2setupRequestIEs_t *)calloc(1,sizeof(*ie));
-  ie->id = E2AP_ProtocolIE_ID_id_E2nodeComponentConfigUpdate;
+  ie->id = E2AP_ProtocolIE_ID_id_E2nodeComponentConfigAddition;
   ie->criticality = E2AP_Criticality_reject;
-  ie->value.present = E2AP_E2setupRequestIEs__value_PR_E2nodeComponentConfigUpdate_List;
+  //ie->value.present = E2AP_E2setupRequestIEs__value_PR_NOTHING;
+  ie->value.present = E2AP_E2setupRequestIEs__value_PR_E2nodeComponentConfigAddition_List;
 
-  e2node_comp_cfg_update_ie = (E2AP_E2nodeComponentConfigUpdate_ItemIEs_t *)calloc(1,sizeof(*e2node_comp_cfg_update_ie));
-  e2node_comp_cfg_update_ie->id = E2AP_ProtocolIE_ID_id_E2nodeComponentConfigUpdate_Item;
+#if 1
+  e2node_comp_cfg_update_ie = (E2AP_E2nodeComponentConfigAddition_ItemIEs_t *)calloc(1,sizeof(*e2node_comp_cfg_update_ie));
+  e2node_comp_cfg_update_ie->id = E2AP_ProtocolIE_ID_id_E2nodeComponentConfigAddition_Item;
   e2node_comp_cfg_update_ie->criticality = E2AP_Criticality_reject;
-  e2node_comp_cfg_update_ie->value.present = E2AP_E2nodeComponentConfigUpdate_ItemIEs__value_PR_E2nodeComponentConfigUpdate_Item;
+  e2node_comp_cfg_update_ie->value.present = E2AP_E2nodeComponentConfigAddition_ItemIEs__value_PR_E2nodeComponentConfigAddition_Item;
   
-  e2NodeCompId = (E2AP_E2nodeComponentID_t *)calloc(1,sizeof(E2AP_E2nodeComponentID_t));
+  e2NodeCompId = &(e2node_comp_cfg_update_ie->value.choice.E2nodeComponentConfigAddition_Item.e2nodeComponentID);
 
   if (e2node_type == E2NODE_TYPE_ENB_CU)
   {
-    e2node_comp_cfg_update_ie->value.choice.E2nodeComponentConfigUpdate_Item.e2nodeComponentType = E2AP_E2nodeComponentType_gNB_CU_UP; //E2AP_E2nodeComponentType_ng_eNB_CU 
-    e2NodeCompId->present = E2AP_E2nodeComponentID_PR_e2nodeComponentTypeGNB_CU_UP;
-    if (asn_umax2INTEGER(&e2NodeCompId->choice.e2nodeComponentTypeGNB_CU_UP.gNB_CU_UP_ID, 100) != 0)
-    	RIC_AGENT_ERROR("gNB_CU_UP_ID encoding failed\n");
+    e2node_comp_cfg_update_ie->value.choice.E2nodeComponentConfigAddition_Item.e2nodeComponentInterfaceType = E2AP_E2nodeComponentInterfaceType_e1; //E2AP_E2nodeComponentType_ng_eNB_CU 
+    e2NodeCompId->present = E2AP_E2nodeComponentID_PR_e2nodeComponentInterfaceTypeE1;
+    if (asn_umax2INTEGER(&e2NodeCompId->choice.e2nodeComponentInterfaceTypeE1.gNB_CU_CP_ID, 100) != 0)
```

Comment:
For the future, can we remove and then parameterize this hardcoded CU-UP ID, like we can configure it in cu.conf/du.conf file?  (at this moment, it should be fine)